src/libstore/derivations.cc

```diff
@@ -1097,36 +1097,52 @@ Derivation Derivation::fromJSON(
     const Store & store,
     const nlohmann::json & json)
 {
+    using nlohmann::detail::value_t;
+
     Derivation res;
 
-    res.name = json["name"];
+    ensureType(json, value_t::object);
 
-    {
-        auto & outputsObj = json["outputs"];
-        for (auto & [outputName, output] : outputsObj.items()) {
+    res.name = valueOfTypeAt(json, value_t::string, "name");
+
+    try {
+        auto & outputsObj = valueOfTypeAt(json, value_t::object, "outputs");
+        for (auto & [outputName, _] : outputsObj.items()) {
+            auto & output = valueOfTypeAt(outputsObj, value_t::object, outputName);
```

Comment:
This should be moved inside `DerivationOutput::fromJSON`. We should not need to re-"at" the field but just the value bound by the `for` loop as before.

Reply:
Good point! I moved the the ensureType call and put the assignment back where it belongs.